src/fmu/dataio/_metadata.py

```diff
@@ -80,276 +63,160 @@ def generate_meta_tracklog() -> list[meta.TracklogEvent]:
-
-    * The FmuProvider, which is typically ERT, which provides a kind of an
-      environment. The FmuProvider may also be legally missing, e.g. when
-      running interactive RMS project
+def _get_objectdata_provider(
+    object: types.Inferrable, dataio: ExportData, meta_existing: dict | None = None
+) -> ObjectDataProvider:
+    """Derive metadata for the object. Reuse metadata if existing"""
+    objdata = objectdata_provider_factory(object, dataio, meta_existing)
+    objdata.derive_metadata()
+    return objdata
+
+
+def _get_filedata_provider(
+    dataio: ExportData, objdata: ObjectDataProvider, fmudata: FmuProvider | None
+) -> FileDataProvider:
+    """Derive metadata for the file."""
+    filedata = FileDataProvider(
+        dataio=dataio,
+        objdata=objdata,
+        rootpath=dataio._rootpath,  # has been updated to case_path if fmurun
+        itername=fmudata.get_iter_name() if fmudata else "",
+        realname=fmudata.get_real_name() if fmudata else "",
+    )
+    filedata.derive_filedata()
+    return filedata
+
+
+def _compute_md5(
+    dataio: ExportData, objdata_extension: str, obj: types.Inferrable
+) -> str:
+    """Compute an MD5 sum using a temporary file."""
+    if objdata_extension.startswith("."):
+        with NamedTemporaryFile(buffering=0, suffix=objdata_extension) as tf:
+            logger.info("Compute MD5 sum for tmp file...: %s", tf.name)
+            return export_file_compute_checksum_md5(
+                obj=obj, filename=Path(tf.name), flag=dataio._usefmtflag
+            )
+    raise ValueError("An extension must start with '.'")
 
-    * The User data, which is a mix of global variables and user set keys
 
-    * The data object itself ie. ObjectDataProvider
+def _reuse_existing_metadata(
+    dataio: ExportData, newmeta: dict, meta_existing: dict
+) -> dict:
+    """Perform a merge procedure if the key `reuse_metadata_rule` is active."""
+    if dataio.reuse_metadata_rule == "preprocessed":
+        return glue_metadata_preprocessed(meta_existing, newmeta.copy())
+    raise ValueError(
+        f"The reuse_metadata_rule {dataio.reuse_metadata_rule} is not supported."
+    )
 
-    Then, metadata are stored as:
 
-    * meta_dollars: holding 'system' state: $schema, version, source
-    * meta_class: a simple string
-    * meta_tracklog: dict of events (datdtime, user)
-    * meta_fmu: nested dict of model, case, etc (complex)
-    * meta_file: dict of paths and checksums
-    * meta_masterdata: dict of (currently) smda masterdata
-    * meta_access: dict with name of field + access rules
-    * meta_objectdata: the data block, may be complex
-    * meta_display: dict of default display settings (experimental)
+def _get_meta_objectdata(objdata: ObjectDataProvider) -> meta.content.AnyContent | None:
+    return (
+        meta.content.AnyContent.model_validate(objdata.metadata)
+        if objdata.metadata["content"] != "unset"
+        else None
+    )
```

Comment:
Was a bit unsure how to handle this `content="unset"`. 
Current implementation is to give a warning if content is not given by the user, and then it will be set to the string "unset". The metadata created would then not be able to upload to Sumo since it has an invalid content string. 

Returning None here if the content is "unset" will set the entire data block in the metadata empty which is not really optimal.. 
I tried creating a pydantic model that would also allow `content ="unset"` but it turned messy since there exists quite a bit of pydantic logic around this content.. And it felt unnecessary since it will still not produce valid metadata and we will in the (hopefully near) future require content as input.